pymongo/mongo_client.py

```diff
@@ -848,6 +851,39 @@ def target():
             self._encrypter = _Encrypter(self, self.__options.auto_encryption_opts)
         self._timeout = options.timeout
 
+        # Add this client to the list of weakly referenced items.
+        # This will be used later if we fork.
+        with MongoClient._clients_lock:
+            MongoClient._clients[self._topology._topology_id] = self
+
+    def _after_fork(self):
```

Comment:
We should rename this method so we can call it in `__init__`, that way we aren't duplicating the Topology and PeriodicExecutor creation code.